### PR TITLE
Tag about files and companions correctly #825

### PR DESCRIPTION
**Pull request by AyanSinhaMahapatra**
Tag about files and LICENSE/NOTICE files accompanying the ABOUT file as "about-mapped", and ignore these files from the scanning step.

### CODE REVIEW

File: scanpipe/models.py [hunk 1]
```diff
@@ -1943,6 +1943,13 @@ def location_path(self):
         path = Path(str(self.path).strip("/"))
         return self.project.codebase_path / path
 
+    @property
+    def name_without_extension(self):
+        """Return the name of the resource without it's extension."""
+        if self.extension:
+            return self.name.replace(self.extension, "")
```
Comment (tdruez) [inline scanpipe/models.py:1950]: Using replace in that context is not ideal.

File: scanpipe/pipes/d2d.py [hunk 2]
```diff
@@ -664,14 +673,19 @@ def map_about_files(project, logger=None):
     for about_file_resource in from_about_files:
         _map_about_file_resource(project, about_file_resource, to_resources)
 
+        about_file_companions = about_file_resource.siblings().filter(
+            name__contains=about_file_resource.name_without_extension
```
Comment (tdruez) [inline scanpipe/pipes/d2d.py:677]: `__startswith` may be more precise. To be confirmed.

File: scanpipe/pipes/d2d.py [hunk 3]
```diff
@@ -664,14 +673,19 @@ def map_about_files(project, logger=None):
     for about_file_resource in from_about_files:
         _map_about_file_resource(project, about_file_resource, to_resources)
 
+        about_file_companions = about_file_resource.siblings().filter(
```
Comment (tdruez) [inline scanpipe/pipes/d2d.py:676]: Shouldn't we limit the QS to specific ABOUT files known extensions?

It seems that there's a risk of selecting unwanted files.

**Review by tdruez — CHANGES_REQUESTED** [review]

File: scanpipe/models.py (hunk 1 above)
Comment (AyanSinhaMahapatra) [reply scanpipe/models.py]: Using `.rstrip()` instead of `replace` now, that should be more precise?
Thanks!

File: scanpipe/pipes/d2d.py (hunk 2 above)
Comment (AyanSinhaMahapatra) [reply scanpipe/pipes/d2d.py]: Yes, this would be more precise, adding this. Thanks!

File: scanpipe/pipes/d2d.py (hunk 3 above)
Comment (AyanSinhaMahapatra) [reply scanpipe/pipes/d2d.py:676]: Yes, added another filter with known extensions. Thanks!

File: scanpipe/models.py (hunk 1 above)
Comment (tdruez) [reply scanpipe/models.py]: `.rstrip` is dangerous:
```
>>> "file.x.ext".rstrip(".ext")
'fil'
```

**Comment by AyanSinhaMahapatra** [comment]
Oh I'm sorry, I thought the commits in SCIO was always one per PR, so I was doing that. I should instead preserve commit history and after approving are the commits merged into one commit?

**Comment by tdruez** [comment]
@AyanSinhaMahapatra we always want to preserve full commit history. It's not possible to review incremental changes without it.
All the branch commits are squashed and merged as 1 commit once the code is approved.

File: scanpipe/models.py (hunk 1 above)
Comment (AyanSinhaMahapatra) [reply scanpipe/models.py]: Yes, using `.rpartition()` instead now. This should be safe.

**Comment by AyanSinhaMahapatra** [comment]
> we always want to preserve full commit history. It's not possible to review incremental changes without it.
All the branch commits are squashed and merged as 1 commit once the code is approved.

I've restored the old commit and added the changes in a separate commit. Will do the same in future. Thanks for the heads-up

File: scanpipe/models.py [hunk 4]
```diff
@@ -1943,6 +1943,14 @@ def location_path(self):
         path = Path(str(self.path).strip("/"))
         return self.project.codebase_path / path
 
+    @property
+    def name_without_extension(self):
+        """Return the name of the resource without it's extension."""
+        if self.extension:
+            name_wout_ext, ext, _ = self.name.rpartition(self.extension)
+            return name_wout_ext
```
Comment (tdruez) [inline scanpipe/models.py:1951]:
```suggestion
            return self.name.rpartition(self.extension)[0]
```

**Review by tdruez — CHANGES_REQUESTED** [review]

File: scanpipe/models.py (hunk 4 above)
Comment (AyanSinhaMahapatra) [reply scanpipe/models.py]: Added this, much cleaner. Thanks!

**Comment by tdruez** [comment]
"AyanSinhaMahapatra force-pushed the update-about-mapping branch from c1c2d1d to dae972c 1 minute ago"

Is the force-push still needed?

**Comment by AyanSinhaMahapatra** [comment]
> Is the force-push still needed?

It was a really minor change, and the last commit that I force-pushed is still really small to be able to review changes without any issues IMHO. The main commit is still intact, I didn't force push both.
I thought the whole point about not force-pushing commits with larger set of changes was to aid review, if instead the issue is to never force push, that's ok too. I can do new commits for every tiny change/test-fix too if that's preferred by you.

**Comment by tdruez** [comment]
@AyanSinhaMahapatra Let's avoid rewriting history as a general rule.
Thanks for the PR 🚀

**Comment by AyanSinhaMahapatra** [comment]
> Let's avoid rewriting history as a general rule.

That makes sense. Will adhere to that in future PRs.
Thanks for the improvement suggestions. :smile:

**Pull request merged**